MapboxGLAndroidSDKTestApp/src/main/java/com/mapbox/mapboxsdk/testapp/activity/location/CustomizedLocationPulsingCircleActivity.java

```diff
@@ -0,0 +1,371 @@
+    setContentView(R.layout.activity_location_layer_customized_pulsing_circle);
+
+    LOCATION_CIRCLE_PULSE_COLOR = Color.BLUE;
+
+    mapView = findViewById(R.id.mapView);
+
+    if (savedInstanceState != null) {
+      lastLocation = savedInstanceState.getParcelable(SAVED_STATE_LOCATION);
+      LOCATION_CIRCLE_PULSE_COLOR = savedInstanceState.getInt(SAVED_STATE_LOCATION_CIRCLE_PULSE_COLOR);
+      LOCATION_CIRCLE_PULSE_DURATION = savedInstanceState.getFloat(SAVED_STATE_LOCATION_CIRCLE_PULSE_DURATION);
+    }
+
+    pulsingCircleDurationButton = findViewById(R.id.button_location_circle_duration);
+    pulsingCircleDurationButton.setText(String.format("%sms",
+        String.valueOf(LOCATION_CIRCLE_PULSE_DURATION)));
+    pulsingCircleDurationButton.setOnClickListener(v -> {
+      if (locationComponent == null) {
+        return;
+      }
+      showDurationListDialog();
+    });
+
+    pulsingCircleColorButton = findViewById(R.id.button_location_circle_color);
+    pulsingCircleColorButton.setOnClickListener(v -> {
+      if (locationComponent == null) {
+        return;
+      }
+      showColorListDialog();
+    });
+
+    mapView.onCreate(savedInstanceState);
+
+    checkPermissions();
+  }
+
+  @SuppressLint("MissingPermission")
+  @Override
+  public void onMapReady(@NonNull MapboxMap mapboxMap) {
+    this.mapboxMap = mapboxMap;
+
+    mapboxMap.setStyle(Style.MAPBOX_STREETS, style -> {
+      locationComponent = mapboxMap.getLocationComponent();
+
+      LocationComponentOptions locationComponentOptions =
+          buildLocationComponentOptions(
+              "waterway-label",
+              LOCATION_CIRCLE_PULSE_COLOR,
+              DEFAULT_LOCATION_CIRCLE_PULSE_ALPHA,
+              LOCATION_CIRCLE_PULSE_DURATION,
+              DEFAULT_LOCATION_CIRCLE_PULSE_RADIUS);
+
+      LocationComponentActivationOptions locationComponentActivationOptions =
+          buildLocationComponentActivationOptions(style,locationComponentOptions);
+
+      locationComponent.activateLocationComponent(locationComponentActivationOptions);
+      locationComponent.setLocationComponentEnabled(true);
+      locationComponent.setCameraMode(CameraMode.TRACKING);
+      locationComponent.forceLocationUpdate(lastLocation);
+    });
+  }
+
+  private LocationComponentOptions buildLocationComponentOptions(@Nullable String idOfLayerBelow,
+                                                                 int pulsingCircleColor,
+                                                                 float pulsingCircleAlpha,
+                                                                 float pulsingCircleDuration,
+                                                                 float pulsingCircleMaxRadius
+  ) {
+    currentPulseDuration = pulsingCircleDuration;
+    boolean enablePulsing = true;
+    if (locationComponent  != null && locationComponent.isLocationComponentActivated()) {
+      enablePulsing = locationComponent.isLocationComponentEnabled();
+    }
```

Comment:
We shouldn't try to implement a client-side solution for:
> when we change any of the pulsing options, even if the component is disabled, the animation is started

Instead, the `LocationComponent#startPulsingLocationCircle` should verify the conditions when the style is applied and avoid showing the layer if the component is disabled.

Reply:
Just made a new commit https://github.com/mapbox/mapbox-gl-native-android/pull/172/commits/e7848ab2b9fc230bf6828aece0e0162cde608c2a

> We shouldn't try to implement a client-side solution

Refactored and removed that part from the `CustomizedLocationPulsingCircleActivity` https://github.com/mapbox/mapbox-gl-native-android/pull/172/commits/e7848ab2b9fc230bf6828aece0e0162cde608c2a#diff-7dcce2d0485478de700d188fc32f455aL144-L147

> Instead, the LocationComponent#startPulsingLocationCircle should verify the conditions when the style is applied and avoid showing the layer if the component is disabled.

Whenever `startPulsingLocationCircle()` is run within `LocationComponent.java`, `options.pulseEnabled()` is first checked

https://github.com/mapbox/mapbox-gl-native-android/blob/e7848ab2b9fc230bf6828aece0e0162cde608c2a/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/location/LocationComponent.java#L729-L731


`isEnabled()` is in reference to the overall `LocationComponent`:
https://github.com/mapbox/mapbox-gl-native-android/blob/e7848ab2b9fc230bf6828aece0e0162cde608c2a/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/location/LocationComponent.java#L742-L745